src/Accelerators/NNPA/Conversion/ZHighToZLow/ZHighToZLow.cpp

```diff
@@ -331,18 +331,23 @@ ZMemRefType convertZTensorToMemRefType(OpBuilder b, Type type) {
         res64 = b.getAffineDimExpr(e1) % constExpr64;
       } else if (layout == ZTensorEncodingAttr::DataLayout::_4DS) {
         // for normal
-        // (e4, e3, e2, e1) -> (e4, e3, e2, e1)
+        // (e4, e3, e2, e1)
         // -> (e4, ceil(e1/64), e3, ceil(e2/32), 32, 64)
         // for bidirectional rnn
-        // (e4, e3, e2, e1) -> (e4, 1, e2, e3 * PADDED(e1))
-        // -> (e4, ceil((e3 * PADDED(e1))/64), e3, ceil(e2/32), 32, 64)
-        assert((shape[1] == 1) && "bidirectional lstm/gru not supported yet");
+        // (e4, e3, e2, e1)
+        // -> (e4, ceil((2 * PADDED(e1))/64), e3, ceil(e2/32), 32, 64)
         e4 = 0;
         e3 = 1;
         e2 = 2;
         e1 = 3;
         n = b.getAffineDimExpr(e4);
-        h = b.getAffineDimExpr(e1).floorDiv(constExpr64);
+        if (shape[1] == 1) {
+          h = b.getAffineDimExpr(e1).floorDiv(constExpr64);
+        } else {
```

Comment:
Does this `else` can handle `shape[1] == -1`?

Reply:
4DS layout is used only for output of LSTM/GRU ops, shape[1] specifies the direction dim and should be 1(forward/backward) or 2(bidirectional), and should not be dynamic. It should be managed in a specific way.
I added the limitation at assert statement to check the condition and add the legality check at src/Accelerators/NNPA/C
onversion/ONNXToZHigh/ONNXLegalityCheck.cpp.